tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/cli/dockerCli/core/jobs/GenerateCodesInLocalJob.ts

```diff
@@ -0,0 +1,66 @@
+import { spawn } from 'child_process';
+import { existsSync } from 'fs';
+import * as path from 'path';
+
+import { getChangedPackageDirectory } from '../../../../utils/git';
+import { sdkToRepoMap } from '../constants';
+import { DockerContext } from '../DockerContext';
+import { DockerTaskEngineContext } from '../DockerTaskEngineContext';
+import { BaseJob } from './BaseJob';
+
+export class GenerateCodesInLocalJob extends BaseJob {
+    context: DockerContext;
+
+    constructor(context: DockerContext) {
+        super();
+        this.context = context;
+    }
+
+    public async cloneRepoIfNotExist(sdkRepos: string[]) {
+        for (const sdkRepo of sdkRepos) {
+            if (!existsSync(path.join(this.context.workDir, sdkRepo))) {
```

Comment:
Could you use this?  https://github.com/steveukx/git-js#git-clone

Reply:
sample-git can handle git operation in most times, but it cannot handle some git operations, such as `git config core.fileMode false --replace-all`. So, to avoid to use git command and simple-git in a mix, I prefer to use git command directly.

Reply:
Let's put this as a future task, I don't want to handle git related operations ourselves just because a small number of command is not supported.